statsmodels/tsa/vecm/vecm.py

```diff
@@ -0,0 +1,254 @@
+        R_0 = Delta_y_1_T.dot(M)
+        R_1 = y_minus1.dot(M)
+        return (M, R_0, R_1)
+    
+    def _estimate_vecm_egls(self, diff_lags, deterministic_terms='', r=1):    
+        _ya, y_1_T, _Dya, Delta_y_1_T, y_minus1, Delta_x = self.estimation_matrices(diff_lags, deterministic_terms)
+        K = y_all.shape[0]
+        T = y_1_T.shape[1]
+        
+        Pi_hat, _Gamma_hat, Sigma_u_hat = _ls_Pi_Gamma(Delta_y_1_T, y_minus1, Delta_x, deterministic_terms)
+        #print("Dy1T.shape: "+str(Delta_y_1_T.shape) + ",  Pi_hat.shape: "+str(Pi_hat.shape) + ",  Gamma_hat.shape: "+str(Gamma_hat.shape))
+        #print("yminus1: " + str(y_minus1.shape) + ",  DX.shape: "+str(Delta_x.shape))
+        alpha_hat = Pi_hat[:, :r]
+        
+        M, R_0, R_1 = self._M_and_R_matrices(T, Delta_x, Delta_y_1_T, y_minus1)
+        R_11 = R_1[:r]
+        R_12 = R_1[r:]
+        #print("Sigma_u_hat: \n"+str(Sigma_u_hat))
+        _alpha_Sigma = alpha_hat.T.dot(np.linalg.inv(Sigma_u_hat))
+        beta_hhat = np.linalg.inv(_alpha_Sigma.dot(alpha_hat)).dot(_alpha_Sigma).dot(R_0-alpha_hat.dot(R_11)).dot(R_12.T).dot(np.linalg.inv(R_12.dot(R_12.T))).T
+        beta_hhat = np.vstack((1, beta_hhat))
+        # ? Gamma_hhat necessary / computed via (Delta_y_1_T - alpha_hat.dot(beta_hhat.T).dot(y_minus1)).dot(Delta_x.dot(np.linalg.inv(np.dot(Delta_x,Delta_x.T))))
+        
+        print("alpha")
+        print(alpha_hat.round(3))
+        print("beta")
+        print(beta_hhat.round(3))
+        return {"alpha":alpha_hat, "beta":beta_hhat}
+    
+    def _estimate_vecm_ml(self, diff_lags, deterministic_terms='', r=1):
+        y_all, y_1_T, _Dya, Delta_y_1_T, y_minus1, Delta_x = self.estimation_matrices(diff_lags, deterministic_terms)
+        K = y_all.shape[0]
+        T = y_1_T.shape[1]
+        
+        M, R_0, R_1 = self._M_and_R_matrices(T, Delta_x, Delta_y_1_T, y_minus1)
+        S = np.bmat([[Ri.dot(Rj.T)/T for Rj in [R_0, R_1]] for Ri in [R_0, R_1]])
+        S00 = S[:K, :K]
+        S01 = S[:K, K:]
+        S10 = S[K:, :K]
+        S11 = S[K:, K:]
+        S11_ = scipy.linalg.sqrtm(S11.I)
+
+        v = np.linalg.eig(S11_ * S10 * S00.I * S01 * S11_)[1]
+        m_beta_tilde = (v[:,:r].T * S11_).T
+        # normalize beta tilde such that eye(r) forms the first r rows of it:
+        m_beta_tilde = m_beta_tilde * m_beta_tilde[:r,:r].I 
+        m_alpha_tilde = S01 * m_beta_tilde * (m_beta_tilde.T * S11 * m_beta_tilde).I
+
+        m_Delta_y_1_T = np.matrix(Delta_y_1_T)
+        m_y_minus1 = np.matrix(y_minus1)
+        m_Delta_x = np.matrix(Delta_x)
+        m_Gamma_tilde = (m_Delta_y_1_T - m_alpha_tilde*m_beta_tilde.T*m_y_minus1) * m_Delta_x.T * (m_Delta_x * m_Delta_x.T).I
+        m_temp = (m_Delta_y_1_T - m_alpha_tilde*m_beta_tilde.T*m_y_minus1 - m_Gamma_tilde*m_Delta_x)
+        m_Sigma_u_tilde = m_temp * m_temp.T / T
+        
+        return {"alpha":np.array(m_alpha_tilde), "beta":np.array(m_beta_tilde),
```

Comment:
Not sure about need to return dict.  Standard is to return tuple.  Mostly a personal preference.